### DISCUSSION TITLE
Desktop Cruiser GL crash on Windows

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#40] [Q&A] [answered]
This latest, and the previous build, more-often-than-not will crash upon loading on my Windows 7 machine (usual "proper" Java).  It loads the command-prompt with the usual text but typically loads the map window at around 400px square in the middle of the screen and then freezes requiring a visit to Task Manager to close it.

Rendering Mode is OpenGL - which appears to be the culprit.  I've tried deleting the Registry settings so it's not that.  v3.0.7 is absolutely fine.

**Comment by devemux86** [comment]
Thanks for the report!

Some changes were made in OpenGL to help the Mac (see https://github.com/devemux86/cruiser/discussions/20),
but it seems we need to revert them for Windows. :slightly_smiling_face:

**Comment by devemux86** [comment]
You can make Cruiser start again with Mapsforge with the command:

```console
java -jar cruiser.jar mapsforge
```

**Reply by MotoUKRider** [reply]
> You can make Cruiser start again with Mapsforge with the command:
> 
> ```
> java -jar cruiser.jar mapsforge
> ```

Yeah, I can revert back to that - but then I lose my sexy map theme :(

**Comment by 0709wiwiwi** [comment]
By mapsforge rendering no problem.  I usually do not set nor use OpenGL.
So I made a quick test by OpenGl: Indeed the report is correct.

Concerning the routeimport from Garmin.  (PM was send to Emux)
(A very complex system by extensions....contains and display so an inbuilt
virtual track)
Willy.

Now closing down for this evening.


Op vr 17 feb. 2023 om 21:41 schreef MotoRiderUK ***@***.***>:

> This latest, and the previous build, more-often-than-not will crash upon
> loading on my Windows 7 machine (usual "proper" Java). It loads the
> command-prompt with the usual text but typically loads the map window at
> around 400px square in the middle of the screen and then freezes requiring
> a visit to Task Manager to close it.
>
> Rendering Mode is OpenGL - which appears to be the culprit. I've tried
> deleting the Registry settings so it's not that. v3.0.7 is absolutely fine.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/devemux86/cruiser/discussions/40>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM5INIOC3YYF722GJULYZVLWX7PAXANCNFSM6AAAAAAU73G5ZQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Willy
***@***.***

**Comment by 0709wiwiwi** [comment]
You can also quickly repair the issue by once using version 3.07 or before
;-) if still available.

Op vr 17 feb. 2023 om 21:48 schreef Emux ***@***.***>:

> You can make Cruiser start again with Mapsforge with the command:
>
> java -jar cruiser.jar mapsforge
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/devemux86/cruiser/discussions/40#discussioncomment-5014312>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM5INIKQCNEOERUANENJILLWX7PZFANCNFSM6AAAAAAU73G5ZQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Willy
***@***.***

**Comment by devemux86** [comment]
I work on Linux, so Windows and Mac are difficult to test often. :slightly_smiling_face:

**Comment by devemux86** [comment]
@MotoUKRider @0709wiwiwi 

Can you try with desktop [Cruiser 3.0.10 (Beta)](https://www.dropbox.com/sh/914t158im4b7lxr/AADOeSE1UjaW4FS2PLc7kwAma)?

**Reply by MotoUKRider** [reply]
Loads just fine :)  Windows 7, "Java" Java and 8GB RAM on Dell 790 box.  Although, when closing the program, the command-window remains running and I have to shut java.exe down in the Task Manager.  Sometimes it exits everything gracefully, but 8/10 times it doesn't.

**Reply by devemux86** [reply]
@MotoUKRider thanks for the feedback!

I updated the zip, can you try with desktop [Cruiser 3.0.10 (Beta 2)](https://www.dropbox.com/sh/914t158im4b7lxr/AADOeSE1UjaW4FS2PLc7kwAma)?

**Reply by MotoUKRider** [reply]
Interesting - if I load it, move the map a little and exit, it closes completely.
If I load a route, then clear it, and exit - the command-box remains visible.

Tested each way 4 times and replicates every time.

**Reply by devemux86** [reply]
With the new desktop [Cruiser 3.0.10 (Beta 2)](https://www.dropbox.com/sh/914t158im4b7lxr/AADOeSE1UjaW4FS2PLc7kwAma)?

How do you run it, with `cruiser.bat`?
What happens if you open a console and run:
```console
java -jar cruiser.jar
```

**Reply by MotoUKRider** [reply]
Yes, with the batch file you supplied.  Using the command-line option produces the same results as detailed above, so no difference in behaviour.

**Reply by devemux86** [reply]
@MotoUKRider what Java?

Can you try with [OpenJDK 11](https://adoptium.net/temurin/releases/?version=11)?

**Reply by MotoUKRider** [reply]
Very latest Java update prior to trying OpenJDK - which gives the same issue, and I've found that simply loading a different map also keeps the command-box open.

**Reply by devemux86** [reply]
This is strange, are you using the **Beta 2**?
Did the previous Cruiser work in this case?

**Reply by devemux86** [reply]
> keeps the command-box open

Does the console have anything written?

If you start with `java -jar cruiser.jar` and exit Cruiser, the command line cursor does not appear?

**Reply by MotoUKRider** [reply]
The console shows the computer details. There are some meshbucket tessellation errors which is a result of that water "mesh" code tweak from yesterday so I changed it back to "false" in the XML style.  In that scenario, no errors at all but still no graceful exit.  It's almost like the shutdown/exist sequence isn't "firing" or whatever calls are being made - perhaps the render engine is busy or a thread ongoing?

**Reply by devemux86** [reply]
If you change a setting or map or routing in Cruiser GL, are they saved on exit?
There is a System.exit process in the same place.

Are you sure running the downloaded OpenJDK?
(use full path for java.exe in console)

**Reply by MotoUKRider** [reply]
Yes, map+settings are preserved after I force a shutdown via TaskManager.  Definitely running OpenJDK :)

**Comment by MotoUKRider** [comment]
Hmmmm.  Even with a short/simple route, the "Turn instructions" option always crashes the desktop app. Could this be a remaining OpenGL bug?

**Reply by devemux86** [reply]
Possibly, OpenGL seems very sensitive, I will see what can be done. :slightly_smiling_face: 

In the past there was a similar report for the "Layers" dialog (when tap on area with many markers).
Does the "Layers" dialog open properly now?

**Reply by MotoUKRider** [reply]
Yes, all OK on the Layers dialog thanks.

**Reply by devemux86** [reply]
Thanks for the feedback!

I will apply the same improvement in other dialogs too and it will be available in the next Cruiser update.

**Comment by devemux86** [comment, marked as answer]
It should work properly with [Cruiser 3.0.10](https://github.com/devemux86/cruiser/releases).

**Comment by DolomitesTom** [comment]
Want to try the desktop version. Which Java Runtime is required, 32 Bit or 64 Bit?

**Reply by devemux86** [reply]
@DolomitesTom 

For the answer please see here:
- https://github.com/devemux86/cruiser/discussions/132
